test/src/l2_test.cpp

```diff
@@ -11,123 +11,56 @@ extern "C" {
 #include "l2.h"
 #include "mock_bm_ip.h"
 #include "mock_bm_os.h"
+#include "mock_bm_adin2111.h"
 }
 
-struct L2Meta {
-  BmL2RxCb rx;
-  BmL2DeviceLinkChangeCb link_change;
-  void *device_handle;
-};
-
-#define num_devices 3
-#define num_port 6
-#define port_per_device (num_port / num_devices)
+#define port_per_device 2
 #define egress_port_offset 51
 
-#if (num_port % num_devices != 0)
-#error "num_devices must divide evenly into num_port"
-#endif
-
 class L2 : public ::testing::Test {
 public:
   rnd_gen RND;
-  static struct L2Meta META[num_devices];
   static uint8_t init_count;
 
-  static BmErr init_cb(void *device_handle, BmL2RxCb rx_cb,
-                       BmL2DeviceLinkChangeCb link_change_cb,
-                       uint8_t port_mask) {
-    BmErr err = BmOK;
-    EXPECT_NE(device_handle, nullptr);
-    EXPECT_NE(rx_cb, nullptr);
-    EXPECT_NE(link_change_cb, nullptr);
-    META[init_count].rx = rx_cb;
-    META[init_count].link_change = link_change_cb;
-    META[init_count].device_handle = device_handle;
-    init_count++;
-    return err;
-  }
-
   static BmErr tx_cb(void *device_handle, uint8_t *buf, uint16_t size,
                      uint8_t port_mask, uint8_t port_offset) {
-    BmErr err = BmOK;
-    bool found = false;
-    for (uint8_t i = 0; i < num_devices; i++) {
-      if (device_handle == META[i].device_handle) {
-        found = true;
-      }
-    }
-    EXPECT_EQ(found, true);
-    return err;
-  }
-
-  static BmErr power_cb(const void *device_handle, bool on, uint8_t port_mask) {
-    BmErr err = BmOK;
-    bool found = false;
-    for (uint8_t i = 0; i < num_devices; i++) {
-      if (device_handle == META[i].device_handle) {
-        found = true;
-      }
-    }
-    EXPECT_EQ(found, true);
-    return err;
+    return BmOK;
   }
 
 private:
 protected:
   L2() {}
   ~L2() override {}
 
-  /*!
-    @brief Create a configured number of devices and initialize driver
-  */
+  NetworkDevice network_device;
+
   void SetUp() override {
-    BmNetDevCfg cfg[num_devices];
+    netdevice_enable_fake.return_val = BmOK;
+    netdevice_disable_fake.return_val = BmOK;
+    network_device = create_mock_network_device();
```

Comment:
Here's how we get and use a mock network device. The `bm_adin2111_test` exercises the real driver internals and thus intentionally expects errors in tests, but everything else should use the mock to avoid trying to do SPI communications in unit tests.